controller/controller.go

```diff
@@ -133,8 +133,8 @@ func New(config Config) (*Controller, error) {
 	return c, nil
 }
 
-func (c *Controller) Boot() {
-	ctx := context.Background()
+func (c *Controller) Boot(ctx context.Context) {
```

Comment:
While you are doing that it would be cool to also add error as a return value.

```suggestion
func (c *Controller) Boot(ctx context.Context) error {
```

Reply:
I would like that, but we do not really handle it. Whenever we call boot we do it in a goroutine and the returned error is always ignored. How should that work then?

Reply:
Ack. I don't have great ideas ATM. I'll try to come up with something in scope of https://github.com/giantswarm/giantswarm/issues/4965. Let's roll.